google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITPaginationTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.compute.v1.integration;
+
+import com.google.cloud.compute.v1.*;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITPaginationTest extends BaseTest {
+  private static ZonesClient zonesClient;
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    ZonesSettings zoneSettings = ZonesSettings.newBuilder().build();
+    zonesClient = ZonesClient.create(zoneSettings);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    zonesClient.close();
+  }
+
+  @Test
+  public void testList() {
+    ZonesClient.ListPagedResponse response = zonesClient.list(DEFAULT_PROJECT);
+    boolean presented = Boolean.FALSE;
+    for (Zone element : response.iterateAll()) {
+      if (element.getName().equals(DEFAULT_ZONE)) {
+        presented = Boolean.TRUE;
+      }
+    }
+    Assert.assertTrue(presented);
+  }
+
+  @Test
+  public void testPaginationMaxResults() {
```

Comment:
Can you please add the same test  (with max results) with built-in pagination (the one which just does response.iterateAll())? To make sure that max results is appreciated.

Reply:
Done